### PR TITLE
Add supportedFeatures flag in status reporting

### PR DESCRIPTION
**Pull request by larohra**
## Description
This PR adds provision for "supportedFeatures". CRP requires the `supportedFeatures` flag to be set in the status blob in order to judge carefully whether the current guest agent supports certain features.

Issue # 

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and Travis.CI is passing.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: azurelinuxagent/common/AgentGlobals.py [hunk 1]
```diff
@@ -17,6 +17,60 @@
...
+# too-few-public-methods<R0903> Disabled: This class is used as an Enum
+class FeatureNames(object):  # pylint: disable=R0903
+    """
+    Enum for defining the Feature Names for all internal and CRP features
+    """
+    MultiConfig = "MultipleExtensionsPerHandler"
+    ExtensionTelemetryPipeline = "ExtensionTelemetryPipeline"
+
+
+class AgentFeature(object):
```
Comment (narrieta) [inline azurelinuxagent/common/AgentGlobals.py:29]: we should move this class to its own file

we also need to clarify its intent (and maybe rename it?) "new features that the Linux Guest Agent supports" seems too vague

should we use this only for the features were CRP needs to adjust its behavior depending on what the agent supports?

File: azurelinuxagent/common/AgentGlobals.py [hunk 2]
```diff
@@ -28,10 +82,39 @@ class AgentGlobals(object):
...
+        if feature_name in AgentGlobals.__internal_features:
+            return AgentGlobals.__internal_features[feature_name]
+
+        if feature_name in AgentGlobals.__crp_supported_features:
+            return AgentGlobals.__crp_supported_features[feature_name]
+
+        raise NotImplementedError("Feature with Name: {0} not found".format(feature_name))
+
+    @staticmethod
+    def get_crp_supported_features():
```
Comment (narrieta) [inline azurelinuxagent/common/AgentGlobals.py:106]: get_agent_supported_features? the current name seems to imply it is CRP who supports those features

File: azurelinuxagent/common/AgentGlobals.py [hunk 3]
```diff
@@ -28,10 +82,39 @@ class AgentGlobals(object):
...
+        """
+        List of features that the GuestAgent currently supports (like FastTrack, MultiConfig, etc).
+        We need to send this list as part of Status reporting to inform CRP of all the features the agent supports.
+        :return: Dict containing all CRP supported features by their names
```
Comment (narrieta) [inline azurelinuxagent/common/AgentGlobals.py:110]: let's add description of key and value to the comment

File: azurelinuxagent/common/AgentGlobals.py (hunk 1 above)
Comment (larohra) [reply azurelinuxagent/common/AgentGlobals.py:29]: Sure, any suggestions? 
CrpSupportedAgentFeature? Interface for defining all features that the Linux Guest Agent supports and needs to report back to CRP?

File: azurelinuxagent/common/AgentGlobals.py (hunk 2 above)
Comment (larohra) [reply azurelinuxagent/common/AgentGlobals.py:106]: The idea was exactly that, basically report status for all those features that CRP is aware of and the agent supports too lol. 
How about `get_agent_supported_feature_list_for_crp`?

File: azurelinuxagent/common/AgentGlobals.py (hunk 3 above)
Comment (larohra) [reply azurelinuxagent/common/AgentGlobals.py:110]: Sure will do

File: azurelinuxagent/common/AgentGlobals.py [hunk 4]
```diff
@@ -17,6 +17,60 @@
 # Requires Python 2.6+ and Openssl 1.0+
 
 
+# too-few-public-methods<R0903> Disabled: This class is used as an Enum
+class FeatureNames(object):  # pylint: disable=R0903
+    """
+    Enum for defining the Feature Names for all internal and CRP features
+    """
+    MultiConfig = "MultipleExtensionsPerHandler"
+    ExtensionTelemetryPipeline = "ExtensionTelemetryPipeline"
```
Comment (narrieta) [inline azurelinuxagent/common/AgentGlobals.py:26]: Let's also separate ETP from this list of features. I think the intention of this list was to be used for the features that require interaction between the agent and CRP and ETP doesn't quite fit there

File: azurelinuxagent/common/AgentGlobals.py (hunk 1 above)
Comment (narrieta) [reply azurelinuxagent/common/AgentGlobals.py:29]: SupportedFeature? or just clarify its intent in the doctring

File: azurelinuxagent/common/AgentGlobals.py (hunk 4 above)
Comment (larohra) [reply azurelinuxagent/common/AgentGlobals.py:26]: I mainly added it here to have a central place for all features, but agreed its not exactly the perfect place for it. Will make the changes

Comment (larohra) [reply azurelinuxagent/common/AgentGlobals.py]: Removed ETP

File: azurelinuxagent/common/AgentGlobals.py (hunk 3 above)
Comment (larohra) [reply azurelinuxagent/common/AgentGlobals.py:91]: Done

File: azurelinuxagent/common/AgentGlobals.py (hunk 1 above)
Comment (larohra) [reply azurelinuxagent/common/AgentGlobals.py:28]: Done

File: azurelinuxagent/common/AgentGlobals.py (hunk 2 above)
Comment (larohra) [reply azurelinuxagent/common/AgentGlobals.py:87]: Done

**Review by pgombar — APPROVED** [review]

**Review by narrieta — APPROVED** [review]

**Pull request merged**